fields/types/list/ListType.js

```diff
@@ -185,17 +185,19 @@ list.prototype.updateItem = function (item, data, files, callback) {
 
 	var field = this;
 	var values = this.getValueFromData(data);
-	// Don't update the value when it is undefined
-	if (values === undefined) {
-		return utils.defer(callback);
-	}
-	// Reset the value when null or an empty string is provided
-	if (values === null || values === '') {
+	// Reset the value when no value or null or an empty string is provided
+	if (values === undefined || values === null || values === '') {
```

Comment:
Sorry, I was cheating here to fix two bugs at once.

With the `undefined` case as it is there is no way to empty out the list. The last item will never be deleted because an empty list won't be submitted at all. TextArray, DateArray, and NumberArray all treat undefined the way I've done here.